### PR TITLE
docs: Updated Node.js SDK examples for SDK 0.3.0

### PR DESCRIPTION
**Pull request by vikram-dagger**
Signed-off-by: Vikram Vaswani <vikram@dagger.io>

### CODE REVIEW

File: docs/current/sdk/nodejs/snippets/get-started/step3/build.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { connect } from "@dagger.io/dagger"
 connect(async (client) => {
   // get Node image
   // get Node version
-  const node = client.container().from("node:16").withExec(["node", "-v"])
+  const node = client.container().from("node:16").withExec({ args: ["node", "-v"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step3/build.js]: withExec is taking a requirement args as a array of strings
`withExec(args: string[], opts?: ContainerWithExecOpts)`
No need to use an object ;). The original code is correct.

File: docs/current/sdk/nodejs/snippets/get-started/step3/build.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import Client, { connect } from "@dagger.io/dagger"
 connect(async (client: Client) => {
   // get Node image
   // get Node version
-  const node = client.container().from("node:16").withExec(["node", "-v"])
+  const node = client.container().from("node:16").withExec({ args: ["node", "-v"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step3/build.ts]:
```suggestion
  const node = client.container().from("node:16").withExec(["node", "-v"])
```

File: docs/current/sdk/nodejs/snippets/get-started/step4/build.js [hunk 3]
```diff
@@ -10,19 +10,18 @@ connect(async (client) => {
   const node = client.container().from("node:16")
 
   // mount cloned repository into Node image
-  const runner = client
-    .container({ id: node })
+  const runner = node
     .withMountedDirectory("/src", source)
     .withWorkdir("/src")
-    .withExec(["npm", "install"])
+    .withExec({ args: ["npm", "install"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step4/build.js:16]:
```suggestion
    .withExec(["npm", "install"])
```

File: docs/current/sdk/nodejs/snippets/get-started/step4/build.js [hunk 4]
```diff
@@ -10,19 +10,18 @@ connect(async (client) => {
...
 
   // run tests
-  await runner.withExec(["npm", "test", "--", "--watchAll=false"]).exitCode()
+  await runner.withExec({ args: ["npm", "test", "--", "--watchAll=false"] }).exitCode()
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step4/build.js]: Same for this one
```suggestion
  await runner.withExec(["npm", "test", "--", "--watchAll=false"]).exitCode()
```

File: docs/current/sdk/nodejs/snippets/get-started/step4/build.js [hunk 5]
```diff
@@ -10,19 +10,18 @@ connect(async (client) => {
...
 
   // build application
   // write the build output to the host
   await runner
-    .withExec(["npm", "run", "build"])
+    .withExec({ args: ["npm", "run", "build"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step4/build.js:25]:
```suggestion
    .withExec(["npm", "run", "build"])
```

File: docs/current/sdk/nodejs/snippets/get-started/step4/build.ts [hunk 6]
```diff
@@ -10,19 +10,18 @@ connect(async (client: Client) => {
   const node = client.container().from("node:16")
 
   // mount cloned repository into Node image
-  const runner = client
-    .container({ id: node })
+  const runner = node
     .withMountedDirectory("/src", source)
     .withWorkdir("/src")
-    .withExec(["npm", "install"])
+    .withExec({ args: ["npm", "install"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step4/build.ts:16]:
```suggestion
    .withExec(["npm", "install"])
```

File: docs/current/sdk/nodejs/snippets/get-started/step4/build.ts [hunk 7]
```diff
@@ -10,19 +10,18 @@ connect(async (client: Client) => {
...
 
   // run tests
-  await runner.withExec(["npm", "test", "--", "--watchAll=false"]).exitCode()
+  await runner.withExec({ args: ["npm", "test", "--", "--watchAll=false"]}).exitCode()
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step4/build.ts]:
```suggestion
  await runner.withExec(["npm", "test", "--", "--watchAll=false"]).exitCode()
```

**Review by slumbering — CHANGES_REQUESTED** [review]

File: docs/current/sdk/nodejs/snippets/get-started/step4/build.ts [hunk 8]
```diff
@@ -10,19 +10,19 @@ connect(async (client: Client) => {
...
   // run tests
-  await runner.withExec(["npm", "test", "--", "--watchAll=false"]).exitCode()
+  await runner.withExec({ args: ["npm", "test", "--", "--watchAll=false"]})
+    .exitCode()
 
   // build application
   // write the build output to the host
   await runner
-    .withExec(["npm", "run", "build"])
+    .withExec({ args: ["npm", "run", "build"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step4/build.ts:25]:
```suggestion
    .withExec(["npm", "run", "build"])
```

**Review by slumbering — CHANGES_REQUESTED** [review]

File: docs/current/sdk/nodejs/snippets/get-started/step5/build.js [hunk 9]
```diff
@@ -18,20 +18,21 @@ connect(async (client) => {
     // highlight-end
 
     // mount cloned repository into Node image
-    const runner = client
-      .container({ id: node })
+    const runner = node
       .withMountedDirectory("/src", source)
       .withWorkdir("/src")
-      .withExec(["npm", "install"])
+      .withExec({ args: ["npm", "install"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step5/build.js:24]:
```suggestion
      .withExec(["npm", "install"])
```

File: docs/current/sdk/nodejs/snippets/get-started/step5/build.js [hunk 10]
```diff
@@ -18,20 +18,21 @@ connect(async (client) => {
...
+    await runner
+      .withExec({ args: ["npm", "test", "--", "--watchAll=false"] })
+      .exitCode()
 
     // highlight-start
     // build application using specified Node version
     // write the build output to the host
     await runner
-      .withExec(["npm", "run", "build"])
+      .withExec({ args: ["npm", "run", "build"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step5/build.js:35]:
```suggestion
      .withExec(["npm", "run", "build"])
```

File: docs/current/sdk/nodejs/snippets/get-started/step5/build.js [hunk 11]
```diff
@@ -18,20 +18,21 @@ connect(async (client) => {
...
+    const runner = node
       .withMountedDirectory("/src", source)
       .withWorkdir("/src")
-      .withExec(["npm", "install"])
+      .withExec({ args: ["npm", "install"] })
 
     // run tests
-    await runner.withExec(["npm", "test", "--", "--watchAll=false"]).exitCode()
+    await runner
+      .withExec({ args: ["npm", "test", "--", "--watchAll=false"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step5/build.js:28]:
```suggestion
      .withExec(["npm", "test", "--", "--watchAll=false"])
```

File: docs/current/sdk/nodejs/snippets/get-started/step5/build.ts [hunk 12]
```diff
@@ -18,20 +18,21 @@ connect(async (client: Client) => {
     // highlight-end
 
     // mount cloned repository into Node image
-    const runner = client
-      .container({ id: node })
+    const runner = node
       .withMountedDirectory("/src", source)
       .withWorkdir("/src")
-      .withExec(["npm", "install"])
+      .withExec({ args: ["npm", "install"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step5/build.ts:24]:
```suggestion
      .withExec(["npm", "install"])
```

File: docs/current/sdk/nodejs/snippets/get-started/step5/build.ts [hunk 13]
```diff
@@ -18,20 +18,21 @@ connect(async (client: Client) => {
...
 
     // run tests
-    await runner.withExec(["npm", "test", "--", "--watchAll=false"]).exitCode()
+    await runner
+      .withExec({ args: ["npm", "test", "--", "--watchAll=false"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step5/build.ts:28]:
```suggestion
      .withExec(["npm", "test", "--", "--watchAll=false"])
```

File: docs/current/sdk/nodejs/snippets/get-started/step5/build.ts [hunk 14]
```diff
@@ -18,20 +18,21 @@ connect(async (client: Client) => {
...
+      .exitCode()
 
     // highlight-start
     // build application using specified Node version
     // write the build output to the host
     await runner
-      .withExec(["npm", "run", "build"])
+      .withExec({ args: ["npm", "run", "build"] })
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step5/build.ts:35]:
```suggestion
      .withExec(["npm", "run", "build"])
```

**Review by slumbering — CHANGES_REQUESTED** [review]

**Comment by vikram-dagger** [comment]
No changes needed

**Pull request closed without merging**